app/code/Magento/Tax/Model/TaxRateCollection.php

```diff
@@ -70,6 +70,21 @@ public function loadData($printQuery = false, $logQuery = false)
         return $this;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getColumnValues($colName)
+    {
+        $col = [];
+        $searchCriteria = $this->getSearchCriteria();
+        $searchResults = $this->taxRateRepository->getList($searchCriteria);
+        foreach ($searchResults->getItems() as $taxRate) {
+            $col[] = $this->createTaxRateCollectionItem($taxRate)->getData($colName);
```

Comment:
Are we really needed to call `createTaxRateCollectionItem` in this context. I believe that we can just `getData` by column name from collection item without creation of `DataObject`.

If it's really needed it would be better to have `TaxRateInterface` item instead of `DataObject`

Reply:
@Den4ik 
`createTaxRateCollectionItem` provides some modification of the data and adds some additional keys that are not in `TaxRateInterface`. We need to use this method so that we have the same fields and the same data that the `loadData` method  returns. In this case, we will be sure that nothing will break in the grid.  
I can change `createTaxRateCollectionItem` to return the interface, but if we look in the comments to `Magento\Framework\Data\Collection::getItems()` and `Magento\Framework\Data\Collection::_addItem()` then these methods work with `DataObject`. Technically it works well, because `Magento\Tax\Model\Calculation\Rate` extends `DataObject`.

In this case, I will change the method name to `updateTaxRateCollectionItem` and change the code. If all is well, I can commit it.
![1627476983928](https://user-images.githubusercontent.com/41998275/127325734-0893b280-678a-4567-8d7e-b6874f82a8ae.jpg)

Reply:
@vlmed It's ok, just thinking about custom modules where used `instanceof` check